### PR TITLE
refactor(upgrade): expose modules to pub scope

### PR DESCRIPTION
**Pull request by niladrih**
This is a refactor which moves most of the upgrade-job 'bin' code to the upgrade lib, and expands its visibility scope from 'pub(crate)' to 'pub'.

### CODE REVIEW

File: k8s/upgrade/src/helm/chart.rs [hunk 1]
```diff
@@ -128,178 +128,178 @@ impl HelmValuesCollection for CoreValues {
...
     }
 
     /// This is the value of the deprecated key for log silence configuration.
-    pub(crate) fn deprecated_log_silence_level(&self) -> &str {
+    pub fn deprecated_log_silence_level(&self) -> &str {
         self.base.deprecated_log_silence_level()
     }
 
-    pub(crate) fn jaeger_operator_image_tag(&self) -> &str {
+    pub fn jaeger_operator_image_tag(&self) -> &str {
```
Comment (tiagolobocastro) [inline k8s/upgrade/src/helm/chart.rs]: missing doc comment

File: k8s/upgrade/src/upgrade_path.rs [hunk 2]
```diff
@@ -52,7 +52,7 @@ pub(crate) fn version_from_core_chart_release(chart_name: &str) -> Result<Versio
     })
 }
 
-pub(crate) async fn core_version_from_umbrella_release(
+pub async fn core_version_from_umbrella_release(
```
Comment (tiagolobocastro) [inline k8s/upgrade/src/upgrade_path.rs]: Missing doc comment

**Review by tiagolobocastro — COMMENTED** [review]
Remind me again, is this for the unified plugin or to allow for more binaries?

**Comment by niladrih** [comment]
> Remind me again, is this for the unified plugin or to allow for more binaries?

This is for the unified plugin, which will be a new binary.

File: k8s/upgrade/src/helm/chart.rs (hunk 1 above)
Comment (niladrih) [reply k8s/upgrade/src/helm/chart.rs]: I'll add in all of these missing doc comments.

**Comment by tiagolobocastro** [comment]
> > Remind me again, is this for the unified plugin or to allow for more binaries?
> 
> This is for the unified plugin, which will be a new binary.

Does it need this big public api surface exposed or can we expose minimal api and leave internal details here?

**Comment by niladrih** [comment]
> > > Remind me again, is this for the unified plugin or to allow for more binaries?
> > 
> > This is for the unified plugin, which will be a new binary.
> 
> Does it need this big public api surface exposed or can we expose minimal api and leave internal details here?

Hmm... sure, I'll trim this down. All of this doesn't need to be in pub scope. 👍

**Comment by niladrih** [comment]
@tiagolobocastro -- I've removed 2nd and 3rd level APIs and only exposed 1st level APIs. I've also hidden APIs which we've decided to not carry over to the unified upgrade, by design.

**Review by sinhaashish — APPROVED** [review]

**Review by tiagolobocastro — APPROVED** [review]

**Comment by niladrih** [comment]
bors merge

**Comment by niladrih** [comment]
bors merge

**Pull request merged**